consensus/src/consensus_observer/pending_blocks.rs

```diff
@@ -65,10 +65,19 @@ impl PendingOrderedBlocks {
         );
 
         // Insert the ordered block into the pending ordered blocks
-        let last_block_round = ordered_block.blocks.last().unwrap().round();
-        self.pending_ordered_blocks
-            .lock()
-            .insert(last_block_round, (ordered_block, None));
+        if let Some(last_block) = ordered_block.blocks.last() {
```

Comment:
here as well, we maintain the invariant the blocks can't be empty, having this type of defensive programming makes it more confusing than helpful

Reply:
Same as https://github.com/aptos-labs/aptos-core/pull/13992/files#r1680832593